File: frontend/src/types/simulation.ts
import type { RiskAnalysisResult } from "@/types/risk";

export interface SimulationRequest {
  scenario: string;
  zone_id?: string | null;
  worker_count?: number;
  hot_work_permit?: boolean;
  electrical_permit?: boolean;
  maintenance_running?: boolean;
  equipment_health?: number | null;
  weather_condition?: string | null;
}

export interface SimulationResponse {
  scenario: string;
  assumptions: Record<string, string>;
  result: RiskAnalysisResult;
}

File: backend/app/schemas/simulation.py
from pydantic import BaseModel

from app.schemas.risk import RiskAnalysisResult


class SimulationRequest(BaseModel):
    scenario: str
    zone_id: str | None = None
    worker_count: int = 0
    hot_work_permit: bool = False
    electrical_permit: bool = False
    maintenance_running: bool = False
    equipment_health: float | None = None
    weather_condition: str | None = None


class SimulationResponse(BaseModel):
    scenario: str
    assumptions: dict[str, str]
    result: RiskAnalysisResult
